lms/lib/courseware_search/lms_result_processor.py

```diff
@@ -73,7 +73,14 @@ def get_display_name(category, item_id):
 
         def get_position_name(section, position):
             """ helper to fetch name corresponding to the position therein """
-            pos = int(position)
+            if position is None:
+                return None
+
+            try:
+                pos = int(position.split('_', 1)[0])
+            except (ValueError, TypeError):
+                pos = 1
```

Comment:
So, I ended up changing my mind on this one - anyone who gets the position from the call to ```path_to_location``` may want to do this same thing - so I added a ```navigation_index``` function so that anyone can use it if they like.